File: Kay/Server/utils/security.py
import hashlib


def hash_password(password: str, salt: str) -> str:
    password_bytes = password.encode('utf-8')
    salt_bytes = salt.encode('utf-8')
    return hashlib.pbkdf2_hmac('sha256', password_bytes, salt_bytes, 100000).hex()

File: Kay/Server/utils/security.js
const crypto = require('crypto');

exports.hashPassword = function(password, salt) {
    const passwordBytes = Buffer.from(password, 'utf-8');
    const saltBytes = Buffer.from(salt, 'utf-8');
    return crypto.pbkdf2Sync(passwordBytes, saltBytes, 100000, 64, 'sha256').toString('hex');
};